client/go/outline/vpn/vpn_linux.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpn
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"net"
+	"sync"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline"
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+	gonm "github.com/Wifx/gonetworkmanager/v2"
+)
+
+type linuxVPNConn struct {
+	id     string
+	status Status
+
+	ctx           context.Context
+	cancel        context.CancelFunc
+	wgEst, wgCopy sync.WaitGroup
+
+	tun     io.ReadWriteCloser
+	outline *outline.Device
```

Comment:
Can you give this a more meaninfgul name. 'outline' doesn't really say much. Perhaps remoteIpDevice? (with tun being the local one)

Reply:
What about `proxy`?